src/Core/p2/iEnv.h

```diff
@@ -17,9 +16,15 @@ struct iEnv
     int32 memlvl;
 };
 
+// bringing in xEnv.h breaks the build entirely
+struct xEnvAsset;
```

Comment:
Alright so I am very unsure about this.

What do you want me to do here?

Reply:
xEnv.h includes iEnv.h already - it has to do this because the xEnv struct in xEnv.h uses the complete iEnv type - so including iEnv.h in xEnv.h creates a circular dependency which breaks the build.

What you did here is correct, forward declaring xEnvAsset in iEnv.h since it is only used as a pointer in iEnv.h.